geonotebook/vis/ktile/handler.py

```diff
@@ -0,0 +1,168 @@
+from concurrent.futures import ThreadPoolExecutor
+from datetime import datetime, timedelta
+import json
+
+from ModestMaps.Core import Coordinate
+from notebook.base.handlers import IPythonHandler
+from tornado import concurrent, ioloop
+from tornado import gen
+from tornado import web
+
+from .utils import serialize_config, serialize_layer
+
+
+class KTileAsyncClient(object):
+    __instance = None
+
+    def __new__(cls, *args, **kwargs):
+        if cls.__instance is None:
+            cls.__instance = super(
+                KTileAsyncClient, cls).__new__(cls, *args, **kwargs)
+        return cls.__instance
+
+    def __init__(self):
+        self.executor = ThreadPoolExecutor(max_workers=20)
+        self.io_loop = ioloop.IOLoop.current()
+
+    @concurrent.run_on_executor
+    def getTileResponse(self, layer, coord, extension,
+                        _debug=False, _profile=False):
+        if _debug:
+            from pudb.remote import set_trace; set_trace(term_size=(255, 70)) # noqa
+
+        if _profile:
+            # profile injected by kernprof
+            output = layer.getTileResponse(coord, extension)
+            return output
+
+        return layer.getTileResponse(coord, extension)
+
+
+class KtileHandler(IPythonHandler):
+
+    def check_xsrf_cookie(self):
+        # TODO: Find a way to correctly communicate XSRF secret to
+        #       the kernel so ingest requests can be property authenticated
+        pass
+
+    def initialize(self, ktile_config_manager):
+        self.ktile_config_manager = ktile_config_manager
+
+        try:
+            if self.request.headers["Content-Type"].startswith(
+                    "application/json"):
+                self.request.json = json.loads(self.request.body)
+        except Exception:
+            self.request.json = None
+
+    def post(self, kernel_id):
```

Comment:
This creates (or replaces) a tilestache config for the particular kernel_id. We may need to extend this to provide more fine-grain control in a put method for updating just the cache etc,  but i'd like to wait until we need too (e.g. in a separate PR)